server/game/gamesteps/challenge/chooseAssaultTargets.js

```diff
@@ -0,0 +1,62 @@
+const BaseStep = require('../basestep.js');
+const AssaultKeywordAction = require('../../GameActions/AssaultkeywordAction.js');
+
+class ChooseAssaultTargets extends BaseStep {
+    constructor(game, challenge, assaultCharacters) {
+        super(game);
+        this.challenge = challenge;
+        this.assaultCharacters = assaultCharacters.filter(character => this.hasAssaultTargets(character));
+        this.assaultTargetChosen = false;
+    }
+
+    hasAssaultTargets(character) {
+        return this.challenge.defendingPlayer.anyCardsInPlay(card => this.canAssault(card, this.challenge, character));
+    }
+
+    continue() {
+        //as soon as one assault target has been chosen, stop asking for targets even when multiple assault characters are in the challenge
+        if(this.assaultCharacters.length > 0 && !this.assaultTargetChosen) {
+            let highestPrintedCost = this.assaultCharacters.map(character => character.getPrintedCost()).reduce((highestPrintedCost, printedCost) => {
+                if(printedCost > highestPrintedCost) {
+                    highestPrintedCost = printedCost;
+                }
+                return highestPrintedCost;
+            }, 0);
```

Comment:
This could be more readable by using Math.max (either in the reduce function, or passing the mapped costs, e.g. `Math.max(...this.assaultCharacters.map(....))`

Reply:
that is really helpful and a lot better, i always forget about the ... operator for arrays